realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -996,17 +1013,22 @@ public void removeChangeListeners() {
     /**
      * Notifies all registered listeners.
      */
-    void notifyChangeListeners() {
+    void notifyChangeListeners(boolean syncBeforeNotifying) {
```

Comment:
When called outside, the `syncBeforeNotifying` is always true? worth to only expose as `notifyChangeListeners()` without any param?